src/asset-swapper/utils/market_operation_utils/path_optimizer.ts

```diff
@@ -35,65 +35,276 @@ function nativeOrderToNormalizedAmounts(
     return { input, output };
 }
 
-function calculateOutputFee(
-    side: MarketOperation,
-    sampleOrNativeOrder: DexSample | NativeOrderWithFillableAmounts,
-    outputAmountPerEth: BigNumber,
-    inputAmountPerEth: BigNumber,
-    fees: FeeSchedule,
-): BigNumber {
-    if (isDexSample(sampleOrNativeOrder)) {
-        const { input, output, source, fillData } = sampleOrNativeOrder;
-        const fee = fees[source]?.(fillData).fee || ZERO_AMOUNT;
-        const outputFee = ethToOutputAmount({
-            input,
-            output,
-            inputAmountPerEth,
-            outputAmountPerEth,
-            ethAmount: fee,
-        });
-        return outputFee;
-    } else {
-        const { input, output } = nativeOrderToNormalizedAmounts(side, sampleOrNativeOrder);
-        const fee = fees[ERC20BridgeSource.Native]?.(sampleOrNativeOrder).fee || ZERO_AMOUNT;
-        const outputFee = ethToOutputAmount({
-            input,
-            output,
-            inputAmountPerEth,
-            outputAmountPerEth,
-            ethAmount: fee,
+export interface PathOptimizerContext {
+    side: MarketOperation;
+    chainId: ChainId;
+    feeSchedule: FeeSchedule;
+    neonRouterNumSamples: number;
+    fillAdjustor: FillAdjustor;
+    pathPenaltyOpts: PathPenaltyOpts;
+    inputAmount: BigNumber;
+}
+
+export class PathOptimizer {
+    private side: MarketOperation;
+    private chainId: ChainId;
+    private feeSchedule: FeeSchedule;
+    private neonRouterNumSamples: number;
+    private fillAdjustor: FillAdjustor;
+    private pathPenaltyOpts: PathPenaltyOpts;
+    private inputAmount: BigNumber;
+
+    constructor(context: PathOptimizerContext) {
```

Comment:
instead of defining `PathOptimizerContext` can we just define all of the args inline?
```suggestion
    constructor(context: {
      side: MarketOperation;
      chainId: ChainId;
      feeSchedule: FeeSchedule;
      neonRouterNumSamples: number;
      fillAdjustor: FillAdjustor;
      pathPenaltyOpts: PathPenaltyOpts;
      inputAmount: BigNumber;
}) {
```